infra/modules/azure_app_service_plan_autoscaler/variables.tf

```diff
@@ -16,13 +16,54 @@ variable "autoscale_name" {
 
 variable "target_service" {
   type = object({
-    app_service_name  = optional(string)
-    function_app_name = optional(string)
+    app_service_id       = optional(string)
+    app_service_name     = optional(string)
+    function_app_id      = optional(string)
+    function_app_name    = optional(string)
+    app_service_plan_id  = optional(string)
+    location = optional(string)
   })
 
+  description = <<EOT
+  Configuration for the target service (App Service or Function App) to which the autoscaler will be applied.
+  
+  For existing services:
+  - Provide either 'app_service_name' or 'function_app_name'.
```

Comment:
What about having an app service autoscaler module AND a function autoscaler module instead? you may consider to share most of the code but this configuration.

Reply:
another option is to something like this instead:

```hcl
variable "target_service" {
  type = object({
    app_service = optional(object({
      app_service_id      = string
      app_service_name    = string
      app_service_plan_id = string
      location            = string
    }))
    function_app = optional(object({
      function_app_id   = string
      function_app_name = string
      location          = string
    }))
  })

  validation {
    condition = length([for v in [var.target_service.app_service, var.target_service.function_app] : v if v != null]) == 1
    error_message = "You must define either app_service or function_app, but not both."
  }
}
```